### ISSUE TITLE
Set TTL/x-expires on queue

### ISSUE THREAD

**Issue by KodeZ** [paolo-losi/stormed-amqp#27]
I cannot see a way to set the TTL on a queue. I would like the queue to die after a period if noone reads from it anymore. Is there a known way to set this? I see that auto-delete is an option. Could one be added for x-expires too?

Ref: bottom of: https://www.rabbitmq.com/ttl.html

**Comment by paolo-losi** [comment]
At the moment x-expires is not supported but it should be really straight forward to add support for it

if you look at the implementation of queue_declare you'll see that there is the "arguments" param
to _queue.Declare. that is where x-expires should be set:

```Python
arguments={'x-expires': ...}
```

The thing to do is to expose the arguments param to queue_declare.
Have you got a change to test it out and prepare a pull request?
I can help you out ...

**Comment by KodeZ** [comment]
Yes, I saw that, but then I also found a link to a forked version that does that, and adds a whole lot of comments too. Check this:
https://github.com/paolo-losi/stormed-amqp/compare/master...sjlongland:master
Is not that what needs to be done?

**Comment by KodeZ** [comment]
I have been testing the forked repository, and I am getting some issues with setting x-expires. It should have an integer parameter, but it does noe accept one. If I do send and integer, it fails on .encode, so I just did a hack, and said str(s).encode, but that gives the problem of the parameter being a string again. Any easy fix that you can think of?
